File: julia/max_divers_data.jl
using LinearAlgebra, Random

Random.seed!(364)

n = 20

# _A and _C are internal - you don't need them.
_A = randn(2*n,n)
_C = diagm(0.5*exp.(randn(n)))

Sigma = _C*_A'*_A*_C
Sigma = 0.5*(Sigma + Sigma')
M = ones(n)*0.2
sigma = sqrt.(diag(Sigma))

# PLEASE USE THE FOLLOWING CODE TO PLOT YOUR SOLUTION
# Replace x_max_divers and x_min_variance with your optimal values
# using Plots
# idx = 1:40
# bar(idx[isodd.(idx)]./2, x_max_divers, bar_width=0.5, label="Max diversification")
# bar!(idx[iseven.(idx)]./2, x_min_variance, bar_width=0.5, label="Min variance")

File: python/max_divers_data.py
import numpy as np

rng = np.random.Generator(np.random.MT19937(seed=12345))
n = 20
# _A and _C are internal - you don't need them.
_A = rng.standard_normal((2*n, n))
_C = np.diag(0.5*np.exp(rng.standard_normal((n,))))

Sigma = _C@_A.T@_A@_C
Sigma = 0.5*(Sigma + Sigma.T)
M = np.ones(n)*0.2
sigma = np.sqrt(np.diag(Sigma))

# USE THE FOLLOWING CODE TO PLOT YOUR SOLUTION
# Replace x_max_divers and x_min_variance with your optimal values
# import matplotlib.pyplot as plt
# plt.bar(np.arange(0,20), x_max_divers, width=0.5, label="Max diversification")
# plt.bar((np.arange(0,20)*2+1)/2,   x_min_variance, width=0.5, label="Min variance")
# plt.legend()